### ISSUE TITLE
Help with Training Inputs

### ISSUE THREAD

**Issue by avichapman** [Cavendish518/SFPNet#7]
Hi.

I am attempting to replicate the training of the SFPNet in order to investigate other training improvements.

Can you please give me some insight on how to do the training? I don't see any example training code in this repo.

Specifically, in the Semantic model's `forward` method, there are three inputs - `input_data`, `xyz` and `batch`.

I assume that `input_data` is a float tensor of shape BxNxF, where the first three in the F dimension are also x, y and z.

Is the `xyz` the voxel coordinates as a long tensor or the original lidar coordinates as a float tensor? Is the shape BxNx3?

Is the batch coordinate just torch.arange(batch_size)? Is the shape B?

Thank you for your help.

Avi

**Comment by Cavendish518** [comment]
Hi,

`input` is SparseConvTensor (check `spconv` lib). `batch` record the ID (due to `spconv` lib) `xyz` is the lidar coordinates (check the code in dataloader and the function related to `xyz` in backbone.py)

For training details, you can refer to [issue #2](https://github.com/Cavendish518/SFPNet/issues/2).

**Comment by avichapman** [comment]
Ah, ok.

So in training mode for a single entry in a minibatch, this would be `torch.arange([number of samples])`?
And in evaluation mode, this would be set in `collation_fn_voxelmean_tta` using the `inds_recons` value returned from the loader?
